src/Network.cpp

```diff
@@ -685,15 +680,18 @@ void innerproduct(const std::vector<float>& input,
                 &input[0], 1,
                 0.0f, &output[0], 1);
 
-    const auto lambda_ReLU = [](auto val) { return (val > 0.0f) ?
-                                            val : 0.0f; };
-
-    for (unsigned int o = 0; o < outputs; o++) {
-        auto val = biases[o] + output[o];
-        if (ReLU) {
+    if (ReLU) {
```

Comment:
What's the point of pulling this out if this is a template argument in the first place? You now duplicate the code in the source, whereas the compiler has everything it needs to know to eliminate the redundant code?